packages/strapi-admin/ee/admin/containers/Roles/CreatePage/Permissions/README.md

```diff
@@ -0,0 +1,377 @@
+              children: [ // This corresponds to the fields that we will display
+                {
+                  label: 'string',
+                  value: 'string',
+                  required: 'boolean', // This key is optional,
+                  children: [], // This key is optional, if it exists it means that a field has nested fields
+                }
+              ]
+             },
+          ]
+        }
+      ],
+      actions: [ // Array of actions
+        {
+          label: 'string', // Label of the action ex: Create,
+          actionId: 'string', // id of the action ex: content-manager.explorer.create
+          subjects: [<string>], // Array of subjects (collection type uid) on which the action can be applied
+          applyToProperties: [<string>] // Array of properties (ex: fields or locales) on which an action can be applied
+        }
+      ]
+    }
+  }
+}
+```
+
+---
+
+## Concepts
+
+- Checkbox: has only 1 `checked=true` or `checked=false`.
+
+- Parent checkbox: A **parent checkbox** is a checkbox which the state value depends on the state of its children ones. It means that we cannot directly access its value from the `modifiedData` object.
+  Such checkbox can have multiple states:
+  - someChecked: `true` or `false`,
+  - checked: `true` of `false`
+    > Both states are coupled: if `someChecked=true` then `checked=false`.
+    > In terms of user's interaction when cliking a parent checkbox is will toggle the value of its children.
+
+Ex: given the following data:
+
+```js
+const modifiedData = {
+  address: {
+    create: {
+      fields: { f1: true, f2: true,}
+      locales: { en: false, fr: false}
+    },
+    update: {
+      enabled: true,
+    }
+  }
+}
+```
+
+From the `modifiedData` object we can identify 4 parent checkboxes:
+
+1. `address` which value depends on the values of `address.create` & `address.create.update` here the state will be: `someChecked=true`
+2. `create` which value depends on the values of `address.create.fields` & `address.create.locles` here the state will be: `someChecked=true, checked=true`
+3. `fields` which value depends on the values of `address.create.fields.f1` & `address.create.fields.f2` here the state will be: `checked=true`
+4. `locales` which value depends on the values of `address.create.locales.en` & `address.create.locales.fr` here the state will be `checked=false, someChecked=falses`
+
+> `address.update` is not a parent checkbox since we can access its value directly `address.update.enabled`
+
+---
+
+## Components architecture
+
+### `<Permissions />` archictecture
+
+```js
+<PermissionsDataManagerProvider>
+  <Tabs>
+    <ContentTypes /> // Used with the `layout.sections.collectionTypes` data
+    <ContentTypes /> // Used with the `layout.sections.singleTypes` data
+    <PluginsAndSettings /> // Used with the `layout.sections.settings` data
+    <PluginsAndSettings /> // Used with the `layout.sections.plugins` data
```

Comment:
According to the comments, the same component is not accessing the same keys? How is it working?

Reply:
We pass the layout prop.


```js
<Tab>
        <ContentTypes
          allContentTypesAttributes={allCollectionTypesAttributes}
          contentTypes={collectionTypes}
        />
        <ContentTypes
          allContentTypesAttributes={allSingleTypesAttributes}
          contentTypes={singleTypes}
        />
        <PluginsAndSettingsPermissions
          permissionType="plugin"
          pluginsPermissionsLayout={pluginsPermissionsLayout}
        />
        <PluginsAndSettingsPermissions
          permissionType="settings"
          pluginsPermissionsLayout={settingsPermissionsLayout}
        />
</Tab>
```

Reply:
it's hardcoded in the DOM